tedge/package/src/scripts/post-install

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+set -e
+
+SERVICE_NAME=tedge-mgmt-server
+
+# Step 1, decide if we should use SystemD or init/upstart
+use_systemctl="True"
+install_service="True"
+systemd_version=0
+if ! command -V systemctl >/dev/null 2>&1; then
+  use_systemctl="False"
+
+  if ! command -V service >/dev/null 2>&1; then
+    install_service="False"
+  fi
+else
+    systemd_version=$(systemctl --version | head -1 | sed 's/systemd //g' | cut -d' ' -f1)
+fi
+
+cleanup() {
+    # This is where you remove files that were not needed on this platform / system
+    if [ "${use_systemctl}" = "False" ]; then
+        rm -f "/usr/lib/systemd/system/${SERVICE_NAME}.service"
+    fi
+
+    if [ "${install_service}" = "False" ]; then
+        rm -f "/etc/chkconfig/${SERVICE_NAME}"
+        rm -f "/etc/init.d/${SERVICE_NAME}"
+    fi
+}
+
+cleanInstall() {
+    printf "\033[32m Post Install of a clean install\033[0m\n"
+
+    if [ "$install_service" = "False" ]; then
+      printf "\033[32m No service manager detected\033[0m\n"
+      return
+    fi
+
+    # Step 3 (clean install), enable the service in the proper way for this platform
+    if [ "${use_systemctl}" = "False" ]; then
+        printf "\033[32m Installing init.d/open-rc service\033[0m\n"
+
+        if command -V chkconfig >/dev/null 2>&1; then
+          chkconfig --add "$SERVICE_NAME"
+        fi
+
+        service "$SERVICE_NAME" restart ||:
+    else
+        printf "\033[32m Installing systemd service\033[0m\n"
+        # rhel/centos7 cannot use ExecStartPre=+ to specify the pre start should be run as root
+        # even if you want your service to run as non root.
+        if [ "${systemd_version%%.*}" -lt 231 ]; then
+            printf "\033[31m systemd version %s is less then 231, fixing the service file \033[0m\n" "${systemd_version}"
+            sed -i "s/=+/=/g" "/usr/lib/systemd/system/${SERVICE_NAME}.service"
+        fi
+        printf "\033[32m Reload the service unit from disk\033[0m\n"
+        systemctl daemon-reload ||:
+        printf "\033[32m Unmask the service\033[0m\n"
+        systemctl unmask "$SERVICE_NAME" ||:
+        printf "\033[32m Set the preset flag for the service unit\033[0m\n"
+        systemctl preset "$SERVICE_NAME" ||:
+        printf "\033[32m Set the enabled flag for the service unit\033[0m\n"
+        systemctl enable "$SERVICE_NAME" ||:
+        systemctl restart "$SERVICE_NAME" ||:
+    fi
+}
+
+upgrade() {
+    printf "\033[32m Post Install of an upgrade\033[0m\n"
+    cleanInstall
+}
+
+unzipContent() {
+    printf "\033[32m Unzip content\033[0m\n"
+    tar -xzvf /opt/tedge-mgmt-server/ui/dist/tedge-mgmt-ui.tar.gz -C /opt/tedge-mgmt-server/ui/dist
+    tar -xzvf /opt/tedge-mgmt-server/server/dist/tedge-mgmt-server.tar.gz -C /opt/tedge-mgmt-server/server/dist
+    chown -R tedge:tedge /opt/tedge-mgmt-server
+    chmod 0755 -R  /opt/tedge-mgmt-server
+}
```

Comment:
The server distribution should be included in the package just as plain files (not as a tarball within the package)...generally linux packages are already tarballs...so it is just unnecessary double compression.

Reply:
I removed the compressing & uncompressing, and include the content as is.